helpers/strReplace.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const factory = () => {
+    return function(str, substr, newSubstr, iteration) {
+        if (typeof str !== 'string' || typeof substr !== 'string' || typeof newSubstr !== 'string') {
+            return 'Invalid Input';
```

Comment:
I think you would rather want to see here an Error, rather than a "Invalid Input" line. 
Please take a look at TypeError and maybe passing some arguments there, so it will be easier to debug